### PR TITLE
Add nf-test to nanoplot, and bump version

### LINKED ISSUE #4089: nanoplot: migrate to nf-test
### Is your feature request related to a problem? Please describe

Add nf-test to nanoplot

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dialvarezs**
## PR checklist

Closes #4089

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/nanoplot/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,61 @@
+nextflow_process {
+
+    name "Test Process NANOPLOT"
+    tag "nanoplot"
+    script "modules/nf-core/nanoplot/main.nf"
```
Comment (sateeshperi) [inline modules/nf-core/nanoplot/tests/main.nf.test:5]:
```suggestion
    script "../main.nf"
```

File: modules/nf-core/nanoplot/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    process "NANOPLOT"
+
```
Comment (sateeshperi) [inline modules/nf-core/nanoplot/tests/main.nf.test:7]: missing tags

File: modules/nf-core/nanoplot/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                input[0] = [
+                    [ id:'test' ], // meta map
+                    [ file(params.test_data['sarscov2']['nanopore']['test_sequencing_summary'], checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assert process.success
```
Comment (sateeshperi) [inline modules/nf-core/nanoplot/tests/main.nf.test:22]: enclose all assertions in assertAll()

**Review by sateeshperi — DISMISSED** [review]

File: modules/nf-core/nanoplot/tests/main.nf.test (hunk 2 above)
Comment (dialvarezs) [reply modules/nf-core/nanoplot/tests/main.nf.test:7]: which tags should I include? 🤔

File: modules/nf-core/nanoplot/tests/main.nf.test (hunk 3 above)
Comment (dialvarezs) [reply modules/nf-core/nanoplot/tests/main.nf.test:22]: Like this?
```groovy
assertAll(
    { assert process.success },
    { assert snapshot(
                process.out.txt,
                process.out.versions
             ).match()
    },
    {
        with(process.out.html.get(0)) {
            assert get(1).collect { p -> file(p).getName() }.contains("NanoPlot-report.html")
        }
    } 
)
```

**Review by Emiller88 — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**